hashicorp/dev-portal
# feat: add give feedback button to top nav

## ✅ "Ready for Review" checklist


---

## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

This PR adds a `Give beta feedback` button to the top navigation bar.

## 🤷 Why

- To receive feedback during the beta
- So that we can remove the alpha feedback bar ASAP, as it causes layout challenges

## 🛠️ How

- Uses `ButtonLink` on table and desktop
- Uses a custom component for the icon-only button on mobile
    - This custom component duplicates styles from `button-link` to serve the needs of this PR
    - Longer term, we likely want to align with HDS on icon size & button size options. In short, in [HDS Button](https://design-system-components-hashicorp.vercel.app/components/button), buttons with 36px height usually only support 16px icon size. We're diverging from the HDS spec here to meet the dev-portal-specific design spec.

## 📸 Design Screenshots

<img width="1121" alt="CleanShot 2022-06-02 at 17 46 05@2x" src="https://user-images.githubusercontent.com/4624598/171743766-a39f47da-74a5-4da7-9b16-a0d5614928cd.png">

## 🧪 Testing

- [ ] Visit any page on the site, such as [the home page][preview], [/waypoint][/waypoint], or [/vault/tutorials/getting-started/getting-started-intro][/vault/tutorials/getting-started/getting-started-intro]
    - Ensure the `Give feedback` CTA in the to header looks as expected, on mobile and desktop
    - Test that it links to [the beta feedback form][form-url]
    - Ensure the rest of the top nav continues to look and function as expected, including on mobile

## 💭 Anything else?

Note: another option considered here is passing customizations through to `ButtonLink` via a new `className` prop (eg could set `--padding-top-bottom` and `--padding-left-right`).

However, in past approaches like this, we tend to create brittleness and eventually ossification in any component customized in this way without clear forward intention, as said component has to account for all possible customizations built on top of it through CSS overrides passed through `className`. There are more detailed notes on this rationale in our CSS modules adoption RFC, [[Digital RFC] Establish conventions for style additions with CSS modules](https://docs.google.com/document/d/1GPN1pt1xPktT9lYkGZZZ-o-SkagOZR1WIsU2X250s-Q/edit#heading=h.j4iia1mkqs9o).

With this in mind, the duplicate code here feels safer for future maintainability.

[task]: https://app.asana.com/0/1202114367927919/1202348169653878/f
[preview]: https://dev-portal-git-zsadd-give-feedback-to-nav-hashicorp.vercel.app/
[/waypoint]: https://dev-portal-git-zsadd-give-feedback-to-nav-hashicorp.vercel.app/waypoint
[/vault/tutorials/getting-started/getting-started-intro]: https://dev-portal-git-zsadd-give-feedback-to-nav-hashicorp.vercel.app/vault/tutorials/getting-started/getting-started-intro
[form-url]: https://forms.gle/fnHLuNahLEhjuKvE6
